drivers/bluetooth/hci/rpmsg.c

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <ipm.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/hci_driver.h>
+
+#include <openamp/open_amp.h>
+#include <metal/sys.h>
+#include <metal/device.h>
+#include <metal/alloc.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#define LOG_MODULE_NAME bt_hci_driver
+#include "common/log.h"
+
+#define RPMSG_CMD 0x01
+#define RPMSG_ACL 0x02
+#define RPMSG_SCO 0x03
+#define RPMSG_EVT 0x04
+
+static void bt_rpmsg_rx(u8_t *data, size_t len);
+
+static K_SEM_DEFINE(sync_sem, 0, 1);
+
+static struct device *ipm_tx_handle;
+static struct device *ipm_rx_handle;
+
+/* Configuration defines */
+
+#define SHM_START_ADDR      (0x20010000 + 0x400)
+#define SHM_SIZE            0x7c00
+#define SHM_DEVICE_NAME     "sram0.shm"
+
+#define VRING_COUNT         2
+#define VRING_TX_ADDRESS    (SHM_START_ADDR + SHM_SIZE - 0x400)
+#define VRING_RX_ADDRESS    (VRING_TX_ADDRESS - 0x400)
+#define VRING_ALIGNMENT     4
+#define VRING_SIZE          16
+
+#define VDEV_STATUS_ADDR    0x20010000
```

Comment:
@galak as part of the TSC call we discussed making the hardware configuration a bit more flexible, perhaps splitting it out into a separate file. Are we talking exclusively about these numbers?

Reply:
@kapi-no could you move lines 31-247 approximately into a separate file, named for example `rpmsg_nrf53.c` in the same folder, so that we move the nRF53-specific side of things to its own file?

Reply:
@galak, @carlescufi, I split openAMP related code and configuration to separate nrf53 file